src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -3157,23 +3147,40 @@ MarkAttributeDirty ColorControlServer::SetQuietReportAttribute(QuieterReportingA
  * @brief
  * Function used to set the remaining time based on quiet reporting conditions.
  * It will update the attribute storage and report the attribute if it is determined dirty.
- * The condition on which the attribute must be reported are defined by the set QuieterReportingPolicyFlags
- * of the quietRemainingTime object and the implicit conditions of the QuieterReportingAttribute class
+ * The condition on which the attribute must be reported are :
+ * - When it changes from 0 to any value higher than 10, or
+ * - When it changes, with a delta larger than 10, caused by the invoke of a command, or
+ * - When it changes to 0.
  *
  * @param endpoint: Endpoint of the RemainingTime attribute to set
  * @param newRemainingTime: Value to update the RemainingTime attribute with
  * @return Success in setting the attribute value or the IM error code for the failure.
  */
-Status ColorControlServer::SetQuietReportRemainingTime(EndpointId endpoint, uint16_t newRemainingTime)
-{
-    uint16_t epIndex = getEndpointIndex(endpoint);
-    auto markDirty   = MarkAttributeDirty::kNo;
-    auto now         = System::SystemClock().GetMonotonicTimestamp();
-    // Establish the quiet report condition for the RemainingTime Attribute
-    // The quiet report is by the previously set policies :
-    // - kMarkDirtyOnChangeToFromZero : When the value changes from 0 to any other value and vice versa, or
-    // - kMarkDirtyOnIncrement : When the value increases.
-    if (quietRemainingTime[epIndex].SetValue(newRemainingTime, now) == AttributeDirtyState::kMustReport)
+Status ColorControlServer::SetQuietReportRemainingTime(EndpointId endpoint, uint16_t newRemainingTime, bool isNewTransition)
+{
+    uint16_t epIndex           = getEndpointIndex(endpoint);
+    uint16_t lastRemainingTime = quietRemainingTime[epIndex].value().ValueOr(0);
+    auto markDirty             = MarkAttributeDirty::kNo;
+    auto now                   = System::SystemClock().GetMonotonicTimestamp();
+
+    auto predicate =
+        [isNewTransition, lastRemainingTime](
+            const typename QuieterReportingAttribute<uint16_t>::SufficientChangePredicateCandidate & candidate) -> bool {
+        constexpr uint16_t reportDelta = 10;
+        if (candidate.newValue.Value() == 0 || (candidate.lastDirtyValue.Value() == 0 && candidate.newValue.Value() > reportDelta))
+        {
+            return true;
+        }
+        else if (isNewTransition &&
```

Comment:
So if we just keep sending commands that increase the value by 2, we can make it arbitrarily large without any reporting happening?

Reply:
I guess yes. 

> When it changes, with a delta larger than 10, caused by the invoke of a command, or
How would you express it in spec and then in implementation?

Essentially, the report of RemainingTime right now would occur only the reception of a Command which increases the current Remaining time by more than 10 and when it reaches 0.

Reply:
Maybe it should also be if the remaining time Increases by a value greater than 10 compared to the last reported value

Reply:
I haven't had time to think this through.  Please check with whoever wrote the spec text?

Reply:
The proposed spec text was mainly to catch any major jumps in transition time, but as pointed out it does not cover the case where a client would slowly ramp up the transition time. @bzbarsky-apple if you have any inputs for this it is much appreciated every single iteration of this I have made, have had some edge case where a report might not be transmitted. 😔

Reply:
it should be related to "10 seconds since last time the value was marked as reportable" or such, maybe.

Reply:
In my latest commit, I set the condition for new Commands to report a the remainingTime if there is a delta > 10 between the current RemainingTime or if the new RemainingTime is > the last Reported value + 10.